backend/lib/routes/info.js

```diff
@@ -58,3 +59,29 @@ async function fetchGardenerVersion () {
     throw err
   }
 }
+
+router.route('/shootspec')
+  .get(async (req, res, next) => {
+    try {
+      const user = req.user
+      const spec = await fetchShootSpec(user)
+      res.send({ spec })
+    } catch (err) {
+      next(err)
+    }
+  })
+
+let shootSpec // Cache, TODO: Need to update cache when apiserver gets updated
+
+async function fetchShootSpec (user) {
+  if (shootSpec) {
+    return shootSpec
+  }
+  const client = user.client
+
+  const res = await client.openAPI.get()
```

Comment:
I would not use user's client to fetch the openAPI as we use the result also for other users and instead use the dashboard's client (even though the result would be the same)
Also, we should check (`selfsubjectaccessreview`) if the user can get the openAPI before returning the (cached) result.

Reply:
The question is if and how we want to cache the result. But you are right, we should not return any information fetched with a user token to any other user. I will have a look at the issue

Reply:
Changed it to use the dashboardClient (admin) to fetch the Open API Spec. However, it is not possible to do a selfsubjectaccessreview as it is not a k8s resource (similar to healthz endpoint).
The only requirement is that the user needs to be authorized...

Reply:
Ok I added a access review for the non resource endpoint